src/elements/content-sidebar/versions/VersionsSidebarContainer.js

```diff
@@ -5,43 +5,75 @@
  */
 
 import React from 'react';
+import noop from 'lodash/noop';
 import API from '../../../api';
 import VersionsSidebar from './VersionsSidebar';
 import { withAPIContext } from '../../common/api-context';
 
 type Props = {
     api: API,
     fileId: string,
+    onVersionChange: (versionId?: string) => void,
+    parentPath: string,
+    versionId?: string,
 };
 
 type State = {
     error?: string,
-    versions: Array<any>,
+    isLoading: boolean,
+    versions: Array<BoxItemVersion>,
 };
 
 class VersionsSidebarContainer extends React.Component<Props, State> {
     props: Props;
 
     state: State = {
+        isLoading: true,
         versions: [],
     };
 
+    static defaultProps = {
+        onVersionChange: noop,
+    };
+
     componentDidMount() {
         const { api, fileId } = this.props;
 
         api.getVersionsAPI(false).getVersions(fileId, this.handleFetchVersionsSuccess, this.handleFetchVersionsError);
     }
 
+    componentDidUpdate({ versionId: prevVersionId }: Props) {
+        const { onVersionChange, versionId } = this.props;
+
+        // Forward the current version id that is passed in via the wrapping route
+        if (prevVersionId !== versionId) {
+            onVersionChange(versionId);
+        }
+    }
+
+    componentWillUnmount() {
+        this.props.onVersionChange();
```

Comment:
What's the reason for calling `onVersionChange` while unmounting?

Reply:
It's essentially a shortcut to reset the current version id, since the location no longer contains one once this component is unmounted.